pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1395,6 +1407,48 @@ func (r *ReconcileClusterDeployment) removeClusterDeploymentFinalizer(cd *hivev1
 	return nil
 }
 
+func (r *ReconcileClusterDeployment) releaseCustomization(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) error {
+	if cd.Spec.ClusterPoolRef == nil || cd.Spec.ClusterPoolRef.CustomizationRef == nil {
```

Comment:
This should also short-circuit if the customization finalizer is already gone. Otherwise we end up doing extra work if we re-enter this function potentially causing a hot loop.